lib/waterfoul/boot_rom.rb

```diff
@@ -49,10 +49,19 @@ class BootROM
       0x3E, 0x01, 0xE0, 0x50
     ].freeze
 
+    attr_reader :cartridge
```

Comment:
There could be a third abstraction on top of the boot rom and the cartridge, but I am not sure it is better.
Essentially, the boot rom alone is not enough for booting, a real cartridge is needed as well.